x-pack/plugins/apm/common/rules/apm_rule_types.ts

```diff
@@ -182,20 +182,24 @@ export function formatAnomalyReason({
   measured,
   windowSize,
   windowUnit,
+  anomalyDetectorTypes,
 }: {
   serviceName: string;
   severityLevel: string;
   measured: number;
   windowSize: number;
   windowUnit: string;
+  anomalyDetectorTypes: AnomalyAlertDetectorType[];
 }) {
+  const detectorTypesToString = anomalyDetectorTypes.toString();
   return i18n.translate(
     'xpack.apm.alertTypes.transactionDurationAnomaly.reason',
     {
-      defaultMessage: `{severityLevel} anomaly with a score of {measured} was detected in the last {interval} for {serviceName}.`,
+      defaultMessage: `{severityLevel} anomaly with a score of {measured} and detector type {detectorTypeToString}, was detected in the last {interval} for {serviceName}.`,
       values: {
         serviceName,
         severityLevel,
+        detectorTypesToString,
```

Comment:
nit: Simply inline this, and better to use `join` to get more control over formatting:
```suggestion
        detectorTypes: anomalyDetectorTypes.join(', ')
```